pkg/skaffold/schema/validation/validation.go

```diff
@@ -570,44 +648,59 @@ func validateLogPrefix(lc latestV1.LogsConfig) []error {
 // validateCustomTest
 // - makes sure that command is not empty
 // - makes sure that dependencies.ignore is only used in conjunction with dependencies.paths
-func validateCustomTest(tcs []*latestV1.TestCase) (errs []error) {
-	for _, tc := range tcs {
-		for _, ct := range tc.CustomTests {
+func validateCustomTest(cfg *parser.SkaffoldConfigEntry, tcs []*latestV1.TestCase) (cfgErrs []ErrorWithLocation) {
+	for i, tc := range tcs {
+		for j, ct := range tc.CustomTests {
 			if ct.Command == "" {
-				errs = append(errs, fmt.Errorf("custom test command must not be empty;"))
+				cfgErrs = append(cfgErrs, ErrorWithLocation{
+					Error:    fmt.Errorf("custom test command must not be empty;"),
+					Location: cfg.YAMLInfos.Locate(&cfg.Test[i].CustomTests[j]),
+				})
 				return
 			}
 
 			if ct.Dependencies == nil {
 				continue
 			}
 			if ct.Dependencies.Command != "" && ct.Dependencies.Paths != nil {
-				errs = append(errs, fmt.Errorf("dependencies can use either command or paths, but not both"))
+				cfgErrs = append(cfgErrs, ErrorWithLocation{
+					Error:    fmt.Errorf("dependencies can use either command or paths, but not both"),
+					Location: cfg.YAMLInfos.Locate(&cfg.Test[i].CustomTests[j]),
+				})
 			}
 			if ct.Dependencies.Paths == nil && ct.Dependencies.Ignore != nil {
-				errs = append(errs, fmt.Errorf("customTest has invalid dependencies; dependencies.ignore can only be used in conjunction with dependencies.paths"))
+				cfgErrs = append(cfgErrs, ErrorWithLocation{
+					Error:    fmt.Errorf("customTest has invalid dependencies; dependencies.ignore can only be used in conjunction with dependencies.paths"),
+					Location: cfg.YAMLInfos.Locate(&cfg.Test[i].CustomTests[j]),
+				})
 			}
 		}
 	}
 	return
 }
 
-func wrapWithContext(config *parser.SkaffoldConfigEntry, errs ...error) []error {
+func wrapWithContext(config *parser.SkaffoldConfigEntry, errs ...ErrorWithLocation) []ErrorWithLocation {
 	var id string
 	if config.Metadata.Name != "" {
 		id = fmt.Sprintf("in module %q", config.Metadata.Name)
 	} else {
-		id = fmt.Sprintf("in unnamed config at index %d", config.SourceIndex)
+		id = fmt.Sprintf("in unnamed config[%d]", config.SourceIndex)
 	}
+
 	for i := range errs {
-		errs[i] = errors.Wrapf(errs[i], "source: %s, %s", config.SourceFile, id)
+		if errs[i].Location == nil || errs[i].Location.StartLine == -1 {
+			errs[i].Error = errors.Wrapf(errs[i].Error, "source: %s,%s", config.SourceFile, id)
+			continue
+		}
+		errs[i].Error = errors.Wrapf(errs[i].Error, "source: %s,%s on line %d column %d",
+			config.SourceFile, id, errs[i].Location.StartLine, errs[i].Location.StartColumn)
```

Comment:
put space after these commas to read better on the command line 
```suggestion
		if errs[i].Location == nil || errs[i].Location.StartLine == -1 {
			errs[i].Error = errors.Wrapf(errs[i].Error, "source: %s, %s", config.SourceFile, id)
			continue
		}
		errs[i].Error = errors.Wrapf(errs[i].Error, "source: %s, %s on line %d column %d",
			config.SourceFile, id, errs[i].Location.StartLine, errs[i].Location.StartColumn)
```

Reply:
good idea!  done